hack/api-docs/generate.sh

```diff
@@ -50,4 +50,4 @@ fi
 gendoc::build
 gendoc::exec \
     -api-dir sigs.k8s.io/gateway-api/apis/v1alpha1 \
-    -out-file "$1"
+    -out-file /dev/stdout | sed '-es|&quot;*\*|\&quot;\&ast;|g' > "$1"
```

Comment:
With https://github.com/kubernetes-sigs/gateway-api/pull/549 merged I don't think this and the associated changes to escaping asterisks are necessary anymore. I think it's still worth looking into ways to improve our docs generation, but what we have now is probably good enough for the v0.2.0 release.